src/codegen/dispatch.rs

```diff
@@ -195,18 +223,53 @@ pub(super) fn function_dispatch(
 }
 
 /// Add the dispatch for function given a matched selector
-fn add_dispatch_case(
+fn add_function_dispatch_case(
     cfg_no: usize,
     func_cfg: &ControlFlowGraph,
+    magic: usize,
     argsdata: &Expression,
     argslen: Expression,
     cases: &mut Vec<(Expression, usize)>,
+    contract_no: usize,
     ns: &Namespace,
     vartab: &mut Vartable,
     cfg: &mut ControlFlowGraph,
 ) {
-    let bb = cfg.new_basic_block(format!("function_cfg_{}", cfg_no));
-    cfg.set_basic_block(bb);
+    let entry = cfg.new_basic_block(format!("function_cfg_{}", cfg_no));
+    cfg.set_basic_block(entry);
+
+    // check for magic
+    let magic_ok = cfg.new_basic_block("magic_ok".into());
+    let magic_bad = cfg.new_basic_block("magic_bad".into());
```

Comment:
What kind of "magic" is this?

Reply:
Formally, the question should be "What does this magic number do?"

Reply:
I've updated the comment